File: frontend/src/types/patient.ts
export interface Patient {
  id: string
  firstName: string
  lastName: string
  cedula: string
  birthDate: string
  phone: string
  address: string
  isActive: boolean
  createdAt: string
}

export interface CreatePatientDto {
  firstName: string
  lastName: string
  cedula: string
  birthDate: string
  phone: string
  address: string
}

export interface UpdatePatientDto {
  firstName: string
  lastName: string
  phone: string
  address: string
}

export interface PatientFormData {
  firstName: string
  lastName: string
  cedula: string
  birthDate: string
  phone: string
  address: string
}

File: backend/src/HealthCore.Domain/Entities/Patient.cs
using HealthCore.Domain.Common;
using HealthCore.Domain.Enums;

namespace HealthCore.Domain.Entities;

public class Patient : BaseEntity
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string IdNumber { get; set; } = string.Empty;
    public DateTime BirthDate { get; set; }
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string? Email { get; set; }
    public GenderType? Gender { get; set; }
    public BloodType? BloodType { get; set; }
    public string? Allergies { get; set; }
    public string? EmergencyContactName { get; set; }
    public string? EmergencyContactPhone { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsDeleted { get; set; }

    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public ICollection<MedicalConsultation> MedicalConsultations { get; set; } = new List<MedicalConsultation>();
    public ICollection<Prescription> Prescriptions { get; set; } = new List<Prescription>();
}